### ISSUE TITLE
plan shows a diff for unspecified open distro alert trigger's action id

### ISSUE THREAD

**Issue by tjbaker** [phillbaker/terraform-provider-elasticsearch#102]
I don't include the trigger's action id in my template, yet plan shows it changing from a value to `null` since upgrading to 1.4.4 from an earlier version.  The OpenDistro API docs don't show including the action id in the post body for create or update, so I'm not sure why plan is flagging it as changing now.  I don't recall this happening before.

Doc here....does not inclide `id` for action:
https://opendistro.github.io/for-elasticsearch-docs/docs/alerting/api/#update-monitor

```
              + schema_version = 1
              ~ triggers       = [
                  ~ {
                      ~ actions   = [
                          ~ {
                                destination_id   = "iJOIc3QBRFnSdtslH7TJ"
                              - id               = "KiGZb3UBij_xm5l20ukt" -> null
```

**Comment by tjbaker** [comment]
IDK if this is related to https://github.com/phillbaker/terraform-provider-elasticsearch/issues/93 ?

**Comment by phillbaker** [comment]
Hm, interesting. Can you share a redacted version of the entire monitor?

**Comment by tjbaker** [comment]
Sorry for the delay, here is a redacted test.  `Plan` shows the action `id` changing to null regardless of whether I specify `id` in the json body.  Every run of plan shows diffs for all my monitors which are similarly configured.


```
terraform {
  required_providers {
    elasticsearch = {
      source  = "phillbaker/elasticsearch"
      version = ">1.5.0"
    }
  }
}


provider "elasticsearch" {
  url = "https://xxx.yyy.com"
}

## Destinations ##

resource "elasticsearch_opendistro_destination" "slack_on_call_channel" {
  body = <<EOF
{
  "name": "On Call Channel",
  "schema_version": 1,
  "type": "slack",
  "slack": {
    "url": "https://hooks.slack.com/services/bogus"
  }
}
EOF
}

resource "elasticsearch_opendistro_destination" "sns_prod_topic" {
  body = <<EOF
{
  "name" : "Production SNS Topic",
  "schema_version" : 1,
  "type" : "sns",
  "sns" : {
    "role_arn" : "arn:aws:iam::bogus",
    "topic_arn" : "arn:aws:sns:bogus"
  }
}
EOF
}

## Monitors ##

resource "elasticsearch_opendistro_monitor" "prod_errors" {

  depends_on = [
    elasticsearch_opendistro_destination.slack_on_call_channel,
    elasticsearch_opendistro_destination.sns_prod_topic
  ]

  body = <<EOF
{
  "type" : "monitor",
  "schema_version" : 1,
  "name" : "Production Errors",
  "enabled" : true,
  "schedule" : {
    "period" : {
      "interval" : 5,
      "unit" : "MINUTES"
    }
  },
  "inputs" : [
    {
      "search" : {
        "indices" : [
          "bogus1",
          "bogus2"
        ],
        "query" : {
          "terminate_after" : 1,
          "query" : {
            "bool" : {
              "must" : [
                {
                  "query_string" : {
                    "query" : "level:ERROR",
                    "fields" : [ ],
                    "type" : "best_fields",
                    "default_operator" : "or",
                    "max_determinized_states" : 10000,
                    "enable_position_increments" : true,
                    "fuzziness" : "AUTO",
                    "fuzzy_prefix_length" : 0,
                    "fuzzy_max_expansions" : 50,
                    "phrase_slop" : 0,
                    "escape" : false,
                    "auto_generate_synonyms_phrase_query" : true,
                    "fuzzy_transpositions" : true,
                    "boost" : 1.0
                  }
                },
                {
                  "range" : {
                    "@timestamp" : {
                      "from" : "{{period_end}}||-5m",
                      "to" : "{{period_end}}",
                      "include_lower" : true,
                      "include_upper" : true,
                      "format" : "epoch_millis",
                      "boost" : 1.0
                    }
                  }
                }
              ],
              "adjust_pure_negative" : true,
              "boost" : 1.0
            }
          }
        }
      }
    }
  ],
  "triggers" : [
    {
      "id": "lgOZb3UB96pyyRQv0ppQ",
      "name" : "Errors",
      "severity" : "1",
      "condition" : {
        "script" : {
          "source" : "ctx.results[0].hits.total.value > 0",
          "lang" : "painless"
        }
      },
      "actions" : [
        {
          "name" : "SNS",
          "destination_id" : "${elasticsearch_opendistro_destination.sns_prod_topic.id}",
          "message_template" : {
            "source" : "bogus",
            "lang" : "mustache"
          },
          "throttle_enabled" : false,
          "subject_template" : {
            "source" : "Production Errors",
            "lang" : "mustache"
          }
        },
        {
          "name" : "Slack",
          "destination_id" : "${elasticsearch_opendistro_destination.slack_on_call_channel.id}",
          "message_template" : {
            "source" : "bogus",
            "lang" : "mustache"
          },
          "throttle_enabled" : false,
          "subject_template" : {
            "source" : "Production Errors",
            "lang" : "mustache"
          }
        }
      ]
    }
  ]
}
EOF
}
```

**Comment by phillbaker** [comment]
One alternative to #93 is adding a new computed property that stores the returned json value from the API separately from the user input JSON, that'd allow tracking user changes separately from API defaults.

**Comment by volym3ad** [comment]
any updates here? Does this plan to be fixed in #93 or separately?

**Comment by phillbaker** [comment]
This should be fixed with merging https://github.com/phillbaker/terraform-provider-elasticsearch/pull/164
